misk-feature-testing/src/main/kotlin/misk/feature/testing/FakeFeatureFlags.kt

```diff
@@ -0,0 +1,69 @@
+package misk.feature.testing
+
+import com.google.common.util.concurrent.AbstractIdleService
+import misk.feature.Attributes
+import misk.feature.Feature
+import misk.feature.FeatureFlags
+import misk.feature.FeatureService
+import java.util.concurrent.ConcurrentHashMap
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * In-memory test implementation of [FeatureFlags] that allows flags to be overridden.
+ */
+@Singleton
+class FakeFeatureFlags @Inject constructor() : AbstractIdleService(),
+    FeatureFlags,
+    FeatureService {
+  override fun startUp() {}
+  override fun shutDown() {}
+
+  private val overrides = ConcurrentHashMap<Feature, Any>()
+
+  override fun getBoolean(feature: Feature, token: String, attributes: Attributes): Boolean {
+    return overrides.getOrDefault(feature, false) as Boolean
+  }
+
+  override fun getInt(feature: Feature, token: String, attributes: Attributes): Int {
+    checkNotNull(
+        overrides[feature]) { "Int flag $feature must be overridden with override() before use" }
+    return overrides[feature] as Int
```

Comment:
if you want, you can combine the cast with the nullcheck AND get a free typecheck

```
return overrides[feature] as? Int
  ?: throw IllegalArgumentException("Int flag $feature must be overridden with override() before use")
```